programs/ssl/ssl_server2.c

```diff
@@ -972,25 +1002,37 @@ int ssl_async_resume( void *connection_ctx_arg,
                         ctx->slot, ctx->delay );
         return( MBEDTLS_ERR_SSL_ASYNC_IN_PROGRESS );
     }
-    (void) output_size; /* mbedtls_pk_size lacks this parameter */
-    ret = mbedtls_pk_sign( key_slot->pk,
-                           ctx->md_alg,
-                           ctx->hash, ctx->hash_len,
-                           output, output_len,
-                           connection_ctx->f_rng, connection_ctx->p_rng );
+    if( ctx->md_alg == MBEDTLS_MD_NONE )
```

Comment:
Minor: Please document in the declaration of `ssl_async_operation_context_t` above that `md_alg` is also used to distinguish between decrypt and sign.